.github/workflows/gt4py-precommit.yml

```diff
@@ -0,0 +1,24 @@
+name: Formatting & compliance (pre-commit)
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
```

Comment:
we could rename this, since otherwise it shows as 'build' in the github webinterface as `Formatting & compliance (pre-commit) / build` . Maybe call it pre-commit and remove the pre-commit above in the parantheses in the `name`.